### PR TITLE
Localization refactor map management

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes
Restructure the files and folders for the field data in the localization

- Add folders for each field, containing the maps and the config override
- The names inside the field folder are equal for each field, therefore only the folder name needs to be adapted
- Only load the required maps
- Fix namespace issue with the map server for the viz map

### CODE REVIEW

File: bitbots_localization/config/config.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   # MISC #
   ########
 
-  init_mode: 4
+  init_mode: 0
```
Comment (jaagut) [inline bitbots_localization/config/config.yaml:6]: What are those init_modes? maybe add some documentation

File: bitbots_localization/config/config.yaml [hunk 2]
```diff
@@ -36,13 +36,13 @@
   ###################
 
 
-  particle_number: 500
-  resampling_interval: 15
+  particle_number: 50
```
Comment (jaagut) [inline bitbots_localization/config/config.yaml:39]: Are these params only for simulation or also applicable in RL?

File: bitbots_localization/config/fields/gazebo/config.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+############
+# Feldraum #
+############
+
+# Settings override for the f-031 test field
```
Comment (jaagut) [inline bitbots_localization/config/fields/gazebo/config.yaml:5]: ???

File: bitbots_localization/config/fields/webots/config.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+############
+# Feldraum #
+############
+
+# Settings override for the f-031 test field
```
Comment (jaagut) [inline bitbots_localization/config/fields/webots/config.yaml:5]: ???

File: bitbots_localization/launch/localization.launch [hunk 5]
```diff
@@ -1,16 +1,20 @@
 <launch>
     <!-- define parameters -->
-    <arg name="fieldname" default="feldraum" doc="Loads field settings, e.g. for the feldraum (f-031)." />
+    <arg name="fieldname" default="webots" doc="Loads field settings, e.g. for the feldraum (f-031)." />
```
Comment (jaagut) [inline bitbots_localization/launch/localization.launch:3]:
```suggestion
    <arg name="fieldname" default="feldraum" doc="Loads field settings (feldraum, webots, gazebo)." />
```
The default should be feldraum, we could automatically load webots if `sim:=true`...

File: bitbots_localization/launch/localization.launch [hunk 6]
```diff
@@ -1,16 +1,20 @@
...
+        <rosparam command="load" file="$(find bitbots_localization)/config/fields/$(arg fieldname)/config.yaml"/>
+
+        <param name="fieldname" value="$(arg fieldname)"/>
     </node>
     <!-- load the map for rviz-->
-    <node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(find bitbots_localization)/models/lines_$(arg fieldname).yaml">
+    <node name="field_map_server" pkg="map_server" type="map_server" output="screen" args="$(find bitbots_localization)/config/fields/$(arg fieldname)/map_server.yaml">
         <param name="frame_id" value="map" />
+        <remap from="/map" to="/field/map"/>
+        <remap from="/map_metadata" to="/field/map_metadata"/>
```
Comment (jaagut) [inline bitbots_localization/launch/localization.launch:18]: @timonegk I think `from` and `to` should be flipped.

**Review by jaagut — CHANGES_REQUESTED** [review]
Why did you remove all those `.png` files (e.g. crosses, posts)?

File: bitbots_localization/launch/localization.launch (hunk 6 above)
Comment (Flova) [reply bitbots_localization/launch/localization.launch:18]: nope ^^

**Review by jaagut — APPROVED** [review]

**Review by jaagut — APPROVED** [review]

**Comment by hendrikbrandt** [comment]
field_y of the field size of the gazebo and webots field should be 6 instead of 5

**Pull request merged**